scalalib/overrides-2.11/scala/collection/immutable/NumericRange.scala

```diff
@@ -113,17 +113,22 @@ extends AbstractSeq[T] with IndexedSeq[T] with Serializable {
 
   import NumericRange.defaultOrdering
 
-  override def min[T1 >: T](implicit ord: Ordering[T1]): T =
-    if (ord eq defaultOrdering(num)) {
+    override def min[T1 >: T](implicit ord: Ordering[T1]): T =
+    // We can take the fast path:
+    // - If the Integral of this NumericRange is also the requested Ordering
+    //   (Integral <: Ordering). This can happen for custom Integral types.
+    // - The Ordering is the default Ordering of a well-known Integral type.
+    if ((ord eq num) || defaultOrdering.get(num).exists(ord eq _)) {
       if (num.signum(step) > 0) start
       else last
     } else super.min(ord)
-
+  
   override def max[T1 >: T](implicit ord: Ordering[T1]): T =
-    if (ord eq defaultOrdering(num)) {
-      if (num.signum(step) > 0) last
-      else start
-    } else super.max(ord)
+    // See comment for fast path in min().
+    if ((ord eq num) || defaultOrdering.get(num).exists(ord eq _)) {
+      if (num.signum(step) > 0) start
+      else last
```

Comment:
Shouldn't this be `last` in `if` and `start` in `else`. Seems like a copy-paste error from `min`.